### PR TITLE
feat: expose `audio-state-changed` on `webContents`

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/37352.

Exposes and documents the `audio-state-changed` event on `webContents`. We previously emitted it, but it was a private and undocumented event. This documents and removes the `-` prefix.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Exposes the `audio-state-changed` event on `webContents`.

### CODE REVIEW

File: shell/browser/api/electron_api_web_contents.cc
Comment (deepak1556) [inline shell/browser/api/electron_api_web_contents.cc:1508]: Why the DCHECK ?

File: docs/api/web-contents.md
Comment (deepak1556) [inline docs/api/web-contents.md:593]: Should this use the new event style https://github.com/electron/electron/pull/37085 ?

File: shell/browser/api/electron_api_web_contents.cc
Comment (codebytere) [reply shell/browser/api/electron_api_web_contents.cc:1508]: I figured it'd be good to ensure that when this event is called it's always the same as `IsCurrentlyAudible()` - happy to remove it though

File: docs/api/web-contents.md
Comment (codebytere) [reply docs/api/web-contents.md:593]: good question! I don't feel strongly either way - i did this because the other PR is not yet merged. Happy to either

1) wait until it's merged and rebase/update accordingly
2) update it now

Which would you prefer?

Comment (deepak1556) [reply docs/api/web-contents.md:593]: Question for @nornagon, once https://github.com/electron/electron/pull/37085 gets merged will there be a refactoring of existing events to make them consistent ?

My preference would be for option `1) wait until it's merged and rebase/update accordingly` if there is a planned refactor on the way.

File: shell/browser/api/electron_api_web_contents.cc
Comment (deepak1556) [reply shell/browser/api/electron_api_web_contents.cc:1508]: Hmm, I think the check makes sense if they were happening at the content layer before the event was emitted and given it is already being done here https://source.chromium.org/chromium/chromium/src/+/main:content/browser/web_contents/web_contents_impl.cc;l=2369-2373, seems this `DCHECK` is redundant.

File: docs/api/web-contents.md
Comment (nornagon) [reply docs/api/web-contents.md:593]: I don't know why that's a question for me, I posted that in our #wg-api channel as a question for the WG :)

File: docs/api/web-contents.md
Comment (nornagon) [inline docs/api/web-contents.md:596]: Yeah, I do think this is a good target for using the new style of event! It seems unlikely that this event will grow extra properties, but you never now...
```suggestion
* `event` Event<>
  * `audible` boolean - Whether audio is currently playing.
```

File: shell/browser/api/electron_api_web_contents.cc
Comment (nornagon) [inline shell/browser/api/electron_api_web_contents.cc:1509]: We don't yet have helpers to make emitting custom event objects ergonomic, so this will have to be the somewhat awkward:
```suggestion
  gin::Handle<gin_helper::internal::Event> event =
      gin_helper::internal::Event::New(isolate);
  v8::Local<v8::Object> event_object = event.ToV8().As<v8::Object>();

  gin::Dictionary dict(isolate, event_object);
  dict.Set("audible", audible);
  EmitWithoutEvent("audio-state-changed", event);
```

File: docs/api/web-contents.md
Comment (deepak1556) [reply docs/api/web-contents.md:593]: Sorry, I was trying to defer to your decision on whether this PR should use the new style already since I didn't see a final consensus on the wg-api discussion thread.

File: docs/api/web-contents.md
Comment (ckerr) [inline docs/api/web-contents.md:596]: This feels a little imprecise; it might be good to copy the spirit of the Chromium comments to help guide readers through the use case of children

```diff
- `audible` boolean - Whether audio is currently playing.
+ `audible` boolean - true if one or more frames or child contents are emitting audio.
```

File: docs/api/web-contents.md
Comment (ckerr) [inline docs/api/web-contents.md:598]: Again a little imprecise for the same reason, but ... revising this in a way that still scans well is not obvious to me. For example this would be more correct but feels awkward:

```suggestion
Emitted when the first frame or child content starts playing, or when the last stops playing.
```

If you have a third idea then let's consider that; otherwise, maybe OK as-is

File: docs/api/web-contents.md
Comment (codebytere) [reply docs/api/web-contents.md:640]: done

File: docs/api/web-contents.md
Comment (codebytere) [reply docs/api/web-contents.md:645]: I updated the property description but i think this is probably more understandable as-is 🤔

**Review by deepak1556 — APPROVED** [review]

**Review by jkleinsc — COMMENTED** [review]
API LGTM

**Pull request merged**
